### PR TITLE
 Add OpenAPI support to the cluster registry.

### PR DESCRIPTION
**Pull request by perotinus**
/sig multicluster

Addresses part of #42. A follow-up will add generating a `swagger.json` file that will be checked in.

/cc @pmorie @madhusudancs 
fyi @font

### CODE REVIEW

File: pkg/apis/clusterregistry/v1alpha1/register.go [hunk 1]
```diff
@@ -49,3 +49,8 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 	metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
 	return nil
 }
+
+// Resource takes an unqualified resource and returns a Group qualified GroupResource
```
Comment (perotinus) [inline pkg/apis/clusterregistry/v1alpha1/register.go]: This fixes an issue with `bazel build`ing `pkg/client/...`

File: pkg/apis/clusterregistry/register.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 )
 
 // GroupName is the group name use in this package
-const GroupName = "clusterregistry"
+const GroupName = "clusterregistry.k8s.io"
```
Comment (font) [inline pkg/apis/clusterregistry/register.go]: FYI, I also took care of this in my commit yesterday https://github.com/kubernetes/cluster-registry/pull/65/commits/49a3c800c87ad0b8e094781aa9d3052eb4364490 based on your feedback for #65.

File: pkg/apis/clusterregistry/v1alpha1/register.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ import (
 )
 
 // GroupName is the group name use in this package
-const GroupName = "clusterregistry"
+const GroupName = "clusterregistry.k8s.io"
```
Comment (font) [inline pkg/apis/clusterregistry/v1alpha1/register.go]: See comment above.

Comment (perotinus) [reply pkg/apis/clusterregistry/v1alpha1/register.go]: Thanks!

File: pkg/apis/clusterregistry/register.go (hunk 2 above)
Comment (perotinus) [reply pkg/apis/clusterregistry/register.go]: Oh, sweet! Will this cause you trouble if I commit this now?

Comment (font) [reply pkg/apis/clusterregistry/register.go]: No that's fine. I can rebase.

**Pull request merged**